src/software/networking/threaded_proto_multicast_listener.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <string>
+#include <thread>
+
+#include "software/networking/proto_multicast_listener.h"
+
+template <class ReceiveProto>
+class ThreadedProtoMulticastListener
```

Comment:
do we have a use for `ProtoMulticastListeners/Senders` can we just make them all threaded for sake of consistency?

Reply:
and by doing so, we would rename `ProtoMulticastListener` to `ThreadedProtoMulticastListener`, and just use it everywhere.

Reply:
I can't think of any cases where we _wouldn't_ want a threaded one, will try homogenize.

Reply:
Should I still call them `Threaded`?

Reply:
Hmmm actually the implementation is a lot cleaner if I leave the `Threaded` and non-threaded versions separate. I'll keep it like that for now and replace all the non-threaded usages.

Reply:
done. I build the non-threaded classes into the bazel targets so that their headers couldn't be included accidentally anywhere, but `cc_library` `srcs` cannot accept `.tpp` files, so I had to change those to `.hpp`